<launch>
    <node name="camera" pkg="usb_cam" type="usb_cam_node" output="log">
        <param name="video_device" value="/dev/video0"/>
        <param name="pixel_format" value="mjpeg"/>
        <param name="image_width" value="1280"/>
        <param name="image_height" value="720"/>
        <param name="framerate" value="20"/>
        <param name="camera_info_url" value="file://$(find ros_people_model)/config/sophia10left/ost.yaml"/>
    </node>

    <node name="face_detector" pkg="ros_people_model" type="run_face_detector.py" output="screen">
        <param name="topic_name" value="/camera/image_raw"/>
        <param name="scale" type="double" value="0.5"/>
        <param name="fps" type="double" value="5.0"/>
        <param name="padding" type="double" value="0.1"/>
    </node>

    <node name="frontal_face_detector" pkg="ros_people_model" type="run_frontal_face_detector.py" output="screen">
        <param name="topic_name" value="/vis_dlib_cnn"/>
        <param name="scale" type="double" value="0.3"/>
        <param name="fps" type="double" value="20.0"/>
    </node>

    <node name="emotion_recogniser" pkg="ros_people_model" type="run_emotion_recogniser.py" output="screen"/>
    <!--<node name="eye_state_recogniser" pkg="ros_people_model" type="run_eye_state_recogniser.py" output="screen"/>-->
    <node name="face_id_recogniser" pkg="ros_people_model" type="run_face_id_recogniser.py" output="screen"/>
    <node name="face_landmarks_recogniser" pkg="ros_people_model" type="run_face_landmarks_recogniser.py" output="screen"/>

    <!--<node name="debugger" pkg="ros_people_model" type="run_debugger.py" output="screen"/>-->
</launch>
